## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 450515`
Commit: `ed414247d788c464094e3b54b8b92b88ab737093`

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>sagittarius-scheme</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>sagittarius-scheme</summary>
<pre>[ 49%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/encauth/siv/siv.c.o
[ 49%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/misc/crypt/crypt_find_prng.c.o
[ 50%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/misc/crypt/crypt_prng_descriptor.c.o
[ 50%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/misc/crypt/crypt_prng_is_valid.c.o
[ 50%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/misc/crypt/crypt_register_prng.c.o
[ 50%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/misc/crypt/crypt_unregister_prng.c.o
[ 51%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/prngs/chacha20.c.o
[ 51%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/prngs/fortuna.c.o
[ 51%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/prngs/rng_get_bytes.c.o
[ 51%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/prngs/rc4.c.o
[ 52%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/prngs/rng_make_prng.c.o
[ 52%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/prngs/sober128.c.o
[ 52%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/prngs/sprng.c.o
[ 52%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/prngs/yarrow.c.o
[ 53%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/misc/hkdf/hkdf.c.o
[ 53%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/misc/pkcs12/pkcs12_kdf.c.o
[ 53%] Building C object ext/crypto/libtomcrypt/CMakeFiles/libtomcrypt.dir/src/misc/bcrypt/bcrypt.c.o
[ 53%] Linking C static library liblibtomcrypt.a
[ 53%] Built target libtomcrypt
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/sa/sagittarius-scheme/fix-roundeven-declaration.patch
+++ b/pkgs/by-name/sa/sagittarius-scheme/fix-roundeven-declaration.patch
@@ -1,0 +1,38 @@
+diff --git a/src/bignum.c b/src/bignum.c
+index 1a5553d5a..6bcdb5b6b 100644
+--- a/src/bignum.c
++++ b/src/bignum.c
+@@ -1949,6 +1949,7 @@ static SgObject small_bignum_to_string(SgBignum *b, int radix, int use_upper)
+ }
+ 
+ /* FIXME this is also in number.c */
++#if !defined(__GLIBC__) || !__GLIBC_PREREQ(2, 26)
+ static inline double roundeven(double v)
+ {
+   double r;
+@@ -1966,6 +1967,7 @@ static inline double roundeven(double v)
+   }
+   return r;
+ }
++  #endif
+ /* this is used here */
+ static SgBignum * bignum_expt(SgBignum *b, long exponent);
+ 
+diff --git a/src/roundeven.inc b/src/roundeven.inc
+index 36d3c5535..1a29eefce 100644
+--- a/src/roundeven.inc
++++ b/src/roundeven.inc
+@@ -31,6 +31,7 @@
+ #include <math.h>
+ #include <float.h>
+ 
++#if !defined(__GLIBC__) || !__GLIBC_PREREQ(2, 26)
+ static inline double roundeven(double v)
+ {
+   double r;
+@@ -50,3 +51,5 @@ static inline double roundeven(double v)
+ }
+ 
+ #endif
++
++#endif

--- a/pkgs/by-name/sa/sagittarius-scheme/package.nix
+++ b/pkgs/by-name/sa/sagittarius-scheme/package.nix
@@ -25,7 +25,7 @@ stdenv.mkDerivation (finalAttrs: {
   pname = "sagittarius-scheme";
   version = "0.9.12";
   src = fetchurl {
-    url = "https://bitbucket.org/ktakashi/sagittarius-scheme/downloads/sagittarius-${finalAttrs.version}.tar.gz";
+    url = "https://github.com/ktakashi/sagittarius-scheme/releases/download/v${finalAttrs.version}/sagittarius-${finalAttrs.version}.tar.gz";
     hash = "sha256-w6aQkC7/vKO8exvDpsSsLyLXrm4FSKh8XYGJgseEII0=";
   };
   preBuild = ''
@@ -76,7 +76,7 @@ stdenv.mkDerivation (finalAttrs: {
       -  mostly works O(n)
       -  Replaceable reader
     '';
-    homepage = "https://bitbucket.org/ktakashi/sagittarius-scheme";
+    homepage = "https://github.com/ktakashi/sagittarius-scheme";
     license = lib.licenses.bsd2;
     platforms = lib.platforms.all;
     maintainers = with lib.maintainers; [ abbe ];

--- a/pkgs/by-name/sa/sagittarius-scheme/package.nix
+++ b/pkgs/by-name/sa/sagittarius-scheme/package.nix
@@ -23,17 +23,23 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "sagittarius-scheme";
-  version = "0.9.12";
+  version = "0.9.14";
   src = fetchurl {
     url = "https://github.com/ktakashi/sagittarius-scheme/releases/download/v${finalAttrs.version}/sagittarius-${finalAttrs.version}.tar.gz";
-    hash = "sha256-w6aQkC7/vKO8exvDpsSsLyLXrm4FSKh8XYGJgseEII0=";
+    hash = "sha256-L0ZKCiSbneWe1+czj83x+bCHPZw1uuYSdJwKHp4qTnk=";
   };
+
+  patches = [
+    ./fix-roundeven-declaration.patch
+  ];
+
   preBuild = ''
     # since we lack rpath during build, need to explicitly add build path
     # to LD_LIBRARY_PATH so we can load libsagittarius.so as required to
     # build extensions
     export ${platformLdLibraryPath}="$(pwd)/build"
   '';
+
   nativeBuildInputs = [
     pkg-config
     cmake
@@ -80,5 +86,6 @@ stdenv.mkDerivation (finalAttrs: {
     license = lib.licenses.bsd2;
     platforms = lib.platforms.all;
     maintainers = with lib.maintainers; [ abbe ];
+    broken = stdenv.hostPlatform.isDarwin;
   };
 })

